// Copyright 2023 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package guideline.vision;

// Options for the ImageSegmentationCalculator
message ImageSegmentationCalculatorOptions {
  message FileDescriptor {
    int32 fd = 1;
    int64 length = 2;
    int64 offset = 3;
  }

  message ContentPointer {
    uint64 pointer = 1;
    int64 length = 2;
  }

  message ModelFile {
    oneof type {
      bytes content = 1;
      string file_path = 2;
      FileDescriptor file_descriptor = 3;
      ContentPointer content_pointer = 4;
    }
  }

  ModelFile model_file = 1;

  // If the model has multiple confidence mask outputs (e.g.
  // background/foreground mask), this indicates the index of the mask that
  // will be returned by the calculator.
  uint32 output_mask_index = 2;
}
